sway-core/src/semantic_analysis/type_check_finalization.rs

```diff
@@ -0,0 +1,32 @@
+//! This module handles the process of iterating through the typed AST and finishing the type
+//! checking step, for type checking steps that need to look at information that was computed
+//! from the initial type checked tree.
+
+use sway_error::handler::{ErrorEmitted, Handler};
+
+use crate::Engines;
+
+use super::TypeCheckContext;
+
+// A simple context that is used to finish type checking.
+pub struct TypeCheckFinalizationContext<'cx, 'dx> {
```

Comment:
Please use full names for lifetimes. It can be `'eng` and `'ctx` but it should be descriptive.